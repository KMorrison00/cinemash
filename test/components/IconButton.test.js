import React from 'react';
import IconButton from '../../components/IconButton';
import { TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';

describe('IconButton', () => {
    it('renders correctly', () => {
        const mockOnPress = jest.fn();
        const tree = renderer
            .create(<IconButton onPress={mockOnPress} name="home" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('calls onPress when pressed', () => {
        const mockOnPress = jest.fn();
        const testRenderer = renderer.create(<IconButton onPress={mockOnPress} name="home" />);
        const testInstance = testRenderer.root;

        // Find the TouchableOpacity component and simulate a press
        const touchableOpacity = testInstance.findByType(TouchableOpacity);
        touchableOpacity.props.onPress();

        // Check if onPress was called
        expect(mockOnPress).toHaveBeenCalled();
    });
});