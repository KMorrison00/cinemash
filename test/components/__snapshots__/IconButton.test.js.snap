// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderRadius": 50,
      "elevation": 2,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 15,
      "shadowColor": "black",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 6,
    }
  }
>
  <Text />
</View>
`;
